### ISSUE TITLE
Display a summary of success/failure at the end of a run (like prove)

### ISSUE THREAD

**Issue by kablamo** [RexOps/Rex#691]
I propose the following:

- If a task dies or returns 0 the task has failed.
- Otherwise a task has succeeded.

The summary might look like this if everything was successful:
```
SUMMARY
deploy:product succeeded on all hosts.
Time: 25 seconds
```

The summary might look like this if there were failures:
```
SUMMARY
deploy:product failed on server1, server9, server23
Time: 25 seconds
```

**Comment by ferki** [comment]
@kablamo: it's a good idea! I like that the summary is kept really short.

**Comment by ferki** [comment]
One question though about the return values of the tasks: what is the reason behind treating 0 as failure? Is that a reliable way of detecting failure?

**Comment by ferki** [comment]
It's already fixed on #697
